File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

float letter(string text);
float word(string text);
float sentence(string text);
int main(void)
{
    string text = get_string("Text:");
    float average_letter = 100/ word(text);

    float L = letter(text) * average_letter;


    float S = sentence(text) *average_letter;

    int index = 0.0588 * L - 0.296 * S - 15.8;
    index = index + 1;
    if (index < 1) {
        printf("Before Grade 1...");

    }
    if (index > 16 ) {
        printf("Grade 16+\n");
    }
    if (index > 1 && index <=16){
        printf("Grade %i\n", index);
}
}


float letter(string text)
{
    float letters = 0.0;
    for (int i = 0,n = strlen(text); i < n; ++i ) {
        text[i] = toupper(text[i]);
    }
    for (int j = 0, a = strlen(text); j < a; ++j) {
        if (text[j] >= 'A' && text[j] <= 'Z') {
            letters += 1;
        }




}
return letters;
}

float word(string text)
{
  float words = 0.0;
  for (int d = 0, b = strlen(text); d < b; ++d) {
      if (text[d] == 32) {
          words += 1;
      }


  }
  words += 1;
return words;
}
float sentence(string text)
{
  float sentences = 0.0;
  for (int c = 0, h = strlen(text); c < h; ++c) {
      if (text[c] == 33 || text[c] == 63 || text[c] == 46 ) {
          sentences += 1;
      }


  }
return sentences;
}

File: readability.py
from cs50 import get_string

def main():
    text = get_string("Text:")
    average_letter = 100/word(text)
    L = letter(text) * average_letter
    S = sentence(text) * average_letter
    index = 0.0588 * L - 0.296 * S - 15.8
    diff = index - int(index)
    add = 1 - diff
    if diff < 0.5:
        index -= diff
    if diff >= 0.5:
        index += add

    if index < 1:
        print("Before Grade 1")

    if index > 16:
        print("Grade 16+")

    if index > 1 and index <= 16:
        print("Grade",int(index))


def letter(text):
    letters = 0.0
    text = text.upper()
    for i in range(len(text)):
        if ord(text[i]) >= 65 and ord(text[i]) <= 90:
            letters += 1
    return letters

def word(text):
    words = 0.0
    for i in range(len(text)):
        if ord(text[i]) == 32:
            words += 1
    return words

def sentence(text):
    sentences = 0.0
    for i in range(len(text)):
        if ord(text[i]) == 33 or ord(text[i]) == 63 or ord(text[i]) == 46:
            sentences += 1

    return sentences


main()
